File: app/src/main/java/com/semid/filechooser/MainActivity.java
package com.semid.filechooser;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.GridLayoutManager;

import com.semid.filechooser.databinding.ActivityMainBinding;
import com.semid.library.FileChooser;
import com.semid.library.FileListener;
import com.semid.library.FileModel;
import com.semid.library.enums.ChooseTypeEnum;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;
    private FileAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);

        binding.takePhoto.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View view) {
                Intent intent = new Intent(getApplicationContext(), Test.class);
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                startActivity(intent);

                return false;
            }
        });
        initAdapter();
        init();
    }

    private void initAdapter() {
        adapter = new FileAdapter(getApplicationContext(), new FileAdapter.Listener() {
            @Override
            public void onDelete(FileModel model) {
                FileChooser.getInstance()
                        .deleteFile(model);
            }
        });
        binding.recyclerView.setLayoutManager(new GridLayoutManager(this, 2));
        binding.recyclerView.setAdapter(adapter);
    }

    private void init() {
        FileListener listener = new FileListener() {
            @Override
            public void newFile(ArrayList<FileModel> files, FileModel fileModel) {
                adapter.updateList(files);
            }

            @Override
            public void onChanged(ArrayList<FileModel> files) {

            }

            @Override
            public void deletedFile(boolean isVideo, FileModel fileModel, int position) {
                adapter.removeItem(position);
            }

            @Override
            public void deletedAllFiles() {

            }
        };

        FileChooser chooser = FileChooser.getInstance();
        chooser.addListener(listener);
    }

    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.choosePhoto:
                FileChooser.getInstance()
                        .intent(ChooseTypeEnum.CHOOSE_PHOTO);
                break;
            case R.id.chooseVideo:
                FileChooser.getInstance()
                        .intent(ChooseTypeEnum.CHOOSE_VIDEO,100);
                break;
            case R.id.takePhoto:
                FileChooser.getInstance()
                        .intent(ChooseTypeEnum.TAKE_PHOTO);
                break;
            case R.id.takeVideo:
                FileChooser.getInstance()
                        .intent(ChooseTypeEnum.TAKE_VIDEO);
                break;
        }
    }
}

File: app/src/main/java/com/semid/filechoosersimple/MainActivity.kt
package com.semid.filechoosersimple

import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide
import com.semid.filechooser.FileChooserActivity
import com.semid.filechooser.FileTypeEnum
import com.semid.filechoosersimple.databinding.ActivityMainBinding
import java.io.File

class MainActivity : AppCompatActivity(), View.OnClickListener {
    private val fileChooser = FileChooserActivity(this)

    private val binding by lazy {
        ActivityMainBinding.inflate(layoutInflater)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)

        fileChooser.fileLiveData
            .observe(this) {
                Log.e("file", it.path + "///")
                Log.e("file", File(it.path).length().toString())

                Glide.with(applicationContext)
                    .load(it.path)
                    .into(binding.coverImg)
            }
    }

    override fun onClick(view: View?) {
        when (view) {
            binding.choosePhotoTxt -> fileChooser.requestFile(FileTypeEnum.CHOOSE_PHOTO)
            binding.chooseVideoTxt -> fileChooser.requestFile(FileTypeEnum.CHOOSE_VIDEO)
            binding.takePhotoTxt -> fileChooser.requestFile(FileTypeEnum.TAKE_PHOTO)
            binding.showFragment -> showFragment()
        }
    }

    private fun showFragment() {
        MyDialogFragment().show(
            supportFragmentManager,
            MyDialogFragment::class.simpleName
        )
    }
}